File: typescript/src/api/resources/executionSessions/client/requests/BranchInput.ts

/**
 * @example
 *     {
 *         idempotency_key: "idempotency_key",
 *         source_run_id: "source_run_id"
 *     }
 */
export interface BranchInput {
    idempotency_key: string;
    source_run_id: string;
}

File: ruby/lib/Sikaru/execution_sessions/types/branch_input.rb
# frozen_string_literal: true

module Sikaru
  module ExecutionSessions
    module Types
      class BranchInput < Internal::Types::Model
        field :project_id, -> { String }, optional: false, nullable: false

        field :session_id, -> { String }, optional: false, nullable: false

        field :idempotency_key, -> { String }, optional: false, nullable: false

        field :source_run_id, -> { String }, optional: false, nullable: false
      end
    end
  end
end
